lua/entities/gmod_wire_expression2/core/vector.lua

```diff
@@ -432,16 +414,12 @@ end
 __e2setcost(10)
 
 --- min/max based on vector length - returns shortest/longest vector
-e2function vector min(vector rv1, vector rv2)
-	local length1 = ( rv1[1] * rv1[1] + rv1[2] * rv1[2] + rv1[3] * rv1[3] ) ^ 0.5
-	local length2 = ( rv2[1] * rv2[1] + rv2[2] * rv2[2] + rv2[3] * rv2[3] ) ^ 0.5
-	if length1 < length2 then return rv1 else return rv2 end
+e2function vector min(vector vec1, vector vec2)
+	return vec1:LengthSqr() < vec2:LengthSqr() and vec1 or vec2
 end
 
-e2function vector max(vector rv1, vector rv2)
-	local length1 = ( rv1[1] * rv1[1] + rv1[2] * rv1[2] + rv1[3] * rv1[3] ) ^ 0.5
-	local length2 = ( rv2[1] * rv2[1] + rv2[2] * rv2[2] + rv2[3] * rv2[3] ) ^ 0.5
-	if length1 > length2 then return rv1 else return rv2 end
+e2function vector max(vector vec1, vector vec2)
+	return vec1:LengthSqr() > vec2:LengthSqr() and vec1 or vec2
```

Comment:
These functions are really stupid. Why don't they max/min the components? Is there a different function that does that?

Reply:
Yes, `maxVec` declared right below this

Reply:
Why would you even need that?

Reply:
I could see uses for both.